File: app/src/main/java/com/naval/todonotesapp/adapter/NotesAdapter.java
package com.naval.todonotesapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.naval.todonotesapp.R;
import com.naval.todonotesapp.clicklisteners.ItemClickListener;
import com.naval.todonotesapp.model.Notes;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private List<Notes> notesList;
    private ItemClickListener itemClickListener;

    public NotesAdapter(List<Notes> list,ItemClickListener listener){
        this.itemClickListener = listener;
        this.notesList = list;
    }

    @NonNull
    @Override
    public NotesAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_adapter_layout,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NotesAdapter.ViewHolder holder, int position) {
        final Notes notes = notesList.get(position);

        holder.adapterTextViewTitle.setText(notes.getTitle());
        holder.adapterTextViewDescription.setText(notes.getDescription());
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                itemClickListener.onClick(notes);
            }
        });
    }

    @Override
    public int getItemCount() {
        return notesList.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder{

        TextView adapterTextViewTitle,adapterTextViewDescription;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            adapterTextViewTitle = itemView.findViewById(R.id.adapterTextViewTitle);
            adapterTextViewDescription = itemView.findViewById(R.id.adapterTextViewDescription);
        }
    }
}

File: app/src/main/java/com/naval/todonotesapp/adapter/NotesAdapter.kt
package com.naval.todonotesapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.CompoundButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.naval.todonotesapp.R
import com.naval.todonotesapp.clicklisteners.ItemClickListener
import com.naval.todonotesapp.db.Notes

class NotesAdapter : RecyclerView.Adapter<NotesAdapter.ViewHolder> {
    private val notesList: List<Notes>
    private val itemClickListener: ItemClickListener

    constructor(notesList: List<Notes>, itemClickListener: ItemClickListener) : super() {
        this.notesList = notesList
        this.itemClickListener = itemClickListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).
                            inflate(R.layout.notes_adapter_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val notes = notesList[position]
        holder.adapterTextViewTitle.text = notes.title
        holder.adapterTextViewDescription.text = notes.description
        holder.adapterMarkStatus.isChecked = notes.isTaskCompleted

        holder.itemView.setOnClickListener { itemClickListener.onClick(notes) }
        holder.adapterMarkStatus.setOnCheckedChangeListener(object : CompoundButton.OnCheckedChangeListener {
            override fun onCheckedChanged(buttonView: CompoundButton?, isChecked: Boolean) {
                notes.isTaskCompleted = isChecked
               itemClickListener.onUpdate(notes)
            }
        })
    }

    override fun getItemCount(): Int {
        return notesList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var adapterTextViewTitle: TextView
        var adapterTextViewDescription: TextView
        var adapterMarkStatus:CheckBox
        init {
            adapterTextViewTitle = itemView.findViewById(R.id.adapterTextViewTitle)
            adapterTextViewDescription = itemView.findViewById(R.id.adapterTextViewDescription)
            adapterMarkStatus = itemView.findViewById(R.id.checkBoxAdapterMarkStatus)
        }
    }

}

